crates/matrix-sdk-common/src/linked_chunk/mod.rs

```diff
@@ -97,15 +97,114 @@ pub mod relational;
 mod updates;
 
 use std::{
-    fmt,
+    cmp,
+    fmt::{self, Display},
     marker::PhantomData,
     ptr::NonNull,
-    sync::atomic::{AtomicU64, Ordering},
+    sync::atomic::{self, AtomicU64},
 };
 
 pub use as_vector::*;
+use ruma::{OwnedRoomId, RoomId};
 pub use updates::*;
 
+/// An identifier for a linked chunk; borrowed variant.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum LinkedChunkId<'a> {
+    Room(&'a RoomId),
+    // TODO(bnjbvr): Soon™.
+    // Thread(&'a RoomId, &'a EventId),
+}
+
+impl AsRef<[u8]> for LinkedChunkId<'_> {
+    fn as_ref(&self) -> &[u8] {
+        match self {
+            LinkedChunkId::Room(room_id) => room_id.as_ref(),
+            // ouch, good luck future me for handling thread keys here
+        }
```

Comment:
Probably better to rename `persistent_key()` or something like this and have it return `impl AsRef<[u8]>`, so that the thread variant can return a `String` formed by concatenating the room and thread ids.

(Alternative: for the room variant, return something like `room:{room_id}`, but it's not really necessary)